transforms.py

```diff
@@ -135,3 +195,238 @@ def forward(self, img):
 class RandomResizedCropWithoutResizeV2(transforms_v2.RandomResizedCrop):
     def _transform(self, inpt, params):
         return F_v2.crop(inpt, **params)
+
+
+class WrapCocoDetectionReferenceSampleForTransformsV2:
+    def __init__(self):
+        num_samples = 117_266
+        wrapper_factory = WRAPPER_FACTORIES[datasets.CocoDetection]
+        mock_dataset = SimpleNamespace(ids=list(range(num_samples)))
+        wrapper = wrapper_factory(mock_dataset)
+        self.wrapper = functools.partial(wrapper, num_samples // 2)
```

Comment:
This whole thing is mocking the dataset so we can re-use the dataset wrappers as transform. `num_samples = 117_266` is the number of samples in the filtered COCO dataset so each sample has one annotation. 

In the wrapper, we get passed the `idx` of the sample

https://github.com/pytorch/vision/blob/39712558247e0d806ceb0df384a6258aa744f950/torchvision/datapoints/_dataset_wrapper.py#L149

and use it like this:

https://github.com/pytorch/vision/blob/39712558247e0d806ceb0df384a6258aa744f950/torchvision/datapoints/_dataset_wrapper.py#L284

Since we don't have access to the `idx` here, I chose a proxy. Since `ids` is a list, I used the half of the maximum, to have on average the same performance as doing the correct lookup.